src/actions/Action.ts

```diff
@@ -6,18 +6,30 @@ class Action implements IAction {
   // We're using map, to overwrite existing keys. for example:
   // addParam(w_100).addParam(w_200) should result in w_200. and not w_100,w_200
   params: Map<string, Param> = new Map();
-  delimiter = ',' // {param}{delimiter}{param} for example: `${'w_100'}${','}${'c_fill'}`
+  delimiter = ','; // {param}{delimiter}{param} for example: `${'w_100'}${','}${'c_fill'}`
 
-  toString() {
-    return Array.from(this.params.values()).join(this.delimiter);
+  getSortedParams(): Param[] {
```

Comment:
Future improvement: this and any "util" function should be static or imported from utils

Reply:
I'm even tempted to somehow have `params.sort()`  (we currently use a map, but maybe we could use our own wrapper over it)

Reply:
yeah good idea, lets merge and refactor